src/Altinn.Platform/Altinn.Platform.Events/Tests/TestingServices/RegisterServiceTest.cs

```diff
@@ -0,0 +1,151 @@
+        {
+            _platformSettings = new Mock<IOptions<PlatformSettings>>();
+            _generalSettings = new Mock<IOptions<GeneralSettings>>();
+            _handlerMock = new Mock<HttpMessageHandler>(MockBehavior.Strict);
+            _contextAccessor = new Mock<IHttpContextAccessor>();
+            _accessTokenGenerator = new Mock<IAccessTokenGenerator>();
+        }
+
+        [Fact]
+        public async Task PartyLookup_MatchFound_IdReturned()
+        {
+            // Arrange
+            Party party = new Party
+            {
+                PartyId = 500000,
+                OrgNumber = "897069650",
+            };
+            int expected = 500000;
+            HttpResponseMessage httpResponseMessage = new HttpResponseMessage
+            {
+                StatusCode = HttpStatusCode.OK,
+                Content = new StringContent(JsonSerializer.Serialize(party), Encoding.UTF8, "application/json")
+            };
+
+            HttpRequestMessage actualRequest = null;
+            void SetRequest(HttpRequestMessage request) => actualRequest = request;
+            InitializeMocks(httpResponseMessage, SetRequest);
+
+            HttpClient httpClient = new HttpClient(_handlerMock.Object);
+
+            RegisterService target = new RegisterService(
+                httpClient,
+                _contextAccessor.Object,
+                _accessTokenGenerator.Object,
+                _generalSettings.Object,
+                _platformSettings.Object,
+                new Mock<ILogger<RegisterService>>().Object);
+
+            // Act
+            int actual = await target.PartyLookup("897069650", null);
+
+            // Assert
+            Assert.Equal(expected, actual);
+            _handlerMock.VerifyAll();
+        }
+
+        [Fact]
+        public async Task PartyLookup_ResponseIsNotSuccessful_PlatformExceptioThrown()
+        {
+
+            // Arrange
+
+            HttpResponseMessage httpResponseMessage = new HttpResponseMessage
+            {
+                StatusCode = HttpStatusCode.NotFound,
+                Content = new StringContent(string.Empty)
+            };
+
+            HttpRequestMessage actualRequest = null;
+            void SetRequest(HttpRequestMessage request) => actualRequest = request;
+            InitializeMocks(httpResponseMessage, SetRequest);
+
+            HttpClient httpClient = new HttpClient(_handlerMock.Object);
+
+            RegisterService target = new RegisterService(
+                httpClient,
+                _contextAccessor.Object,
+                _accessTokenGenerator.Object,
+                _generalSettings.Object,
+                _platformSettings.Object,
+                new Mock<ILogger<RegisterService>>().Object);
+
+            // Act & Assert
+            try
+            {
+                int actual = await target.PartyLookup("16069412345", null);
+            }
+            catch (PlatformHttpException e)
+            {
+                Assert.NotNull(e);
```

Comment:
This assert will never fail as e will never be null. 

Another thing to note is that the test will go green if there are no exceptions at all. If the intention is to assert that there were an exception the Assert must be moved outside the catch. Assign the exception here to a new variable that is initialized to be null. Then Assert that it has gotten a value.

Reply:
Refactored now.

Reply:
to assert.Throws